assets/scripts/menus/MenuBar.jsx

```diff
@@ -0,0 +1,170 @@
+    const signInVisibilityStyle = userId
+      ? { display: 'none' } : {}
+
+    // Buttons have `disabled={false}` because
+    // Firefox sometimes disables some buttons… unsure why
+    // --
+    // Note on translations - the {t()} method doesn't re-render when the
+    // language changes, so this still uses the `data-i18n` method to pick up
+    // on text content changes. Individual menus will re-render with {t()}.
+    return (
+      <nav className='menu-bar'>
+        <ul className='menu-bar-left'>
+          <li className='menu-bar-title'>
+            <div className='streetmix-logo' />
+            <h1>Streetmix</h1>
+          </li>
+          <li id='help-menu-item'>
+            <button
+              data-name='help'
+              data-i18n='menu.item.help'
+              className='menu-attached'
+              disabled={false}
+              onClick={this.onClickMenuButton}
+            >
+              Help
+            </button>
+          </li>
+          <li id='contact-menu-item'>
+            <button
+              data-name='contact'
+              data-i18n='menu.item.contact'
+              className='menu-attached'
+              disabled={false}
+              onClick={this.onClickMenuButton}
+            >
+              Contact
+            </button>
+          </li>
+        </ul>
+        <ul className='menu-bar-right'>
+          <li id='identity-menu-item' style={identityMenuVisibilityStyle}>
+            <button
+              data-name='identity'
+              className='menu-attached'
+              disabled={false}
+              onClick={this.onClickMenuButton}
+            >
+              <div className='avatar' data-user-id={userId} />
+              <span className='user-id'>{userId}</span>
+            </button>
+          </li>
+          <li id='sign-in-menu-item' style={signInVisibilityStyle}>
+            <a href={`/${URL_SIGN_IN_REDIRECT}`} className='command' data-i18n='menu.item.sign-in'>
+              Sign in
+            </a>
+          </li>
+          <li>
+            <a href='/new' target='_blank' data-i18n='menu.item.new-street'>
+              New street
+            </a>
+          </li>
+          <li id='gallery-link'>
+            <a href={myStreetsLink} data-i18n='menu.item.my-streets' onClick={onMyStreetsClick}>
+              My streets
+            </a>
+          </li>
+          {
+            (() => {
+              if (debug.experimental) {
+                return (
+                  <li id='settings-menu-item'>
+                    <button
+                      data-name='settings'
+                      data-i18n='menu.item.settings'
+                      className='menu-attached'
+                      disabled={false}
+                      onClick={this.onClickMenuButton}
+                    >
+                      Settings
+                    </button>
+                  </li>
+                )
+              }
+            })()
```

Comment:
@ryross I tried doing this:

```js
const SettingsButton = (
  <li id='settings-menu-item'>
    <button
      data-name='settings'
      data-i18n='menu.item.settings'
      className='menu-attached'
      disabled={false}
      onClick={this.onClickMenuButton}
    >
      Settings
    </button>
  </li>
)

/* ... */

          {
            (() => {
              if (debug.experimental) {
                return {SettingsButton}
              }
            })()
          }
```

Here's my question about how this works -- in the PR currently, the elements that represent `SettingsButton` will only be created if the `if()` statement is true, correct? But if it's pulled out and made into a variable outside of the `return`, we're still re-creating the elements for `SettingsButton` on every render even if it's not being displayed.

Alternatively, could the `if()` statement exist during the variable assignment?

```js

    const SettingsButton = (debug.experimental) ?
      (<li id='settings-menu-item'>
        <button
          data-name='settings'
          data-i18n='menu.item.settings'
          className='menu-attached'
          disabled={false}
          onClick={this.onClickMenuButton}
        >
          Settings
        </button>
      </li>) : null

/* later ... */
          </li>
          {SettingsButton}
          <li id='share-menu-item'>
```